test/Cut.hs

```diff
@@ -22,34 +26,40 @@ tests = testGroup "Cut"
     [ testMonad
     -- , testMonadFix
     , testCut
-    ] >>= ($ RunL CutC.runCutA)
+    ] >>= ($ runL CutC.runCutA)
+  , testGroup "ReaderC · CutC" $
+    Cut.test (local (id @R)) (m (gen0 S.<> Reader.gen0 r) (\ m -> genN m S.<> Reader.genN r m)) a b (pair <*> r <*> unit) (Run (CutC.runCutA . uncurry runReader))
```

Comment:
This is where things start to get really interesting.

We want to understand issues like #279 which only manifest due to relationships between different effects and carriers. In #279 specifically, it only happens when a `cutfail` occurs in a higher-order position in an effect run _after_ `CutC`. So `ReaderC r (CutC PureC)` doesn’t suffer this behaviour, because the `ReaderC` is eliminated first, but `CutC (ReaderC r PureC)` does.

In order to demonstrate this, we need to run the tests separately from `CutC PureC` to accommodate the different handlers and initial state. We also throw `Reader` actions into the generators for good measure, altho this is not technically required to manifest the bug. Finally, we provide a homomorphism for `Cut` actions—in this case `local id`—which the test suite applies to each action. (I’m not particularly thrilled with this approach, as it’s syntactically noisy and extremely manual, but I don’t have a better one at the moment and I wanted to get this in flight before fixing #279.)